src/python/pants/backend/native/tasks/native_external_library_fetch.py

```diff
@@ -0,0 +1,267 @@
+# coding=utf-8
+# Copyright 2018 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
+                        unicode_literals, with_statement)
+
+import os
+import re
+from distutils.dir_util import copy_tree
+
+from pants.backend.native.subsystems.conan import Conan
+from pants.backend.native.targets.external_native_library import ExternalNativeLibrary
+from pants.base.exceptions import TaskError
+from pants.invalidation.cache_manager import VersionedTargetSet
+from pants.task.task import Task
+from pants.util.contextutil import environment_as
+from pants.util.dirutil import safe_mkdir
+from pants.util.memo import memoized_property
+from pants.util.objects import Exactly
+from pants.util.osutil import get_normalized_os_name
+from pants.util.process_handler import subprocess
+
+
+class ConanRequirement(object):
+  """A wrapper class to encapsulate a Conan package requirement."""
+
+  @staticmethod
+  def _translate_conan_pkg_id_to_directory_path(pkg_string):
+    return pkg_string.replace('@', '/')
+
+  @staticmethod
+  def _build_conan_cmdline_args(pkg_spec, os_name=None):
+    os_name = os_name or get_normalized_os_name()
+    conan_os_opt = None
+    if os_name == 'linux':
+      conan_os_opt = 'Linux'
+    elif os_name == 'darwin':
+      conan_os_opt = 'Macos'
+    else:
+      raise ValueError('Unsupported platform: {}'.format(conan_os_opt))
+    args = ['install', pkg_spec]
+    if conan_os_opt:
+      args.extend(['-s', 'os=' + conan_os_opt])
+    return args
+
+  @classmethod
+  def parse(cls, conan_pkg_spec):
+    directory_path = cls._translate_conan_pkg_id_to_directory_path(conan_pkg_spec)
+    fetch_cmdline_args = cls._build_conan_cmdline_args(conan_pkg_spec)
+    return cls(conan_pkg_spec, directory_path, fetch_cmdline_args)
+
+  def __init__(self, pkg_spec, directory_path, fetch_cmdline_args):
+    self.pkg_spec = pkg_spec
+    self.directory_path = directory_path
+    self.fetch_cmdline_args = fetch_cmdline_args
+
+  def parse_conan_stdout_for_pkg_sha(self, stdout):
+    # TODO(cmlivingston): regex this
```

Comment:
This should just be:
```
def parse_conan_stdout_for_pkg_sha(self, stdout):
  pkg_spec_escaped = re.escape(self.pkg_spec)
  pkg_spec_pattern = re.compile('^{}:(.*)$'.format(pkg_spec_escaped), flags=re.MULTILINE)
  return pkg_spec_pattern.search(stdout).group(1)

Reply:
Also, Stu informed me on another PR that the `TODO(name)` style is a remnant of when Pants used to be a smaller project, and that a `TODO` or `FIXME` with a sufficient explanation inline or a link to an issue should be used instead at this point. I tend to use e.g. `TODO(#5998)` when the TODO would be fixed by that issue or PR, but that's purely something I made up -- all that's important is the issue link.

Reply:
This is a pretty tricky output to parse and I am trying to get this merged asap so I will make a follow-up ticket to fix this.

Reply:
Sounds like a plan!